src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -29,14 +33,40 @@
     /**
      * Looks at the definition of {@link AbstractProject} and list up the related github repositories,
      * then puts them into the collection.
+     *
+     * @deprecated Use {@link #parseAssociatedNames(Job, Collection)}
+     */
+    @Deprecated
+    public void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result) {
+        parseAssociatedNames((Job) job, result);
+    }
+
+    /**
+     * Looks at the definition of {@link Job} and list up the related github repositories,
+     * then puts them into the collection.
      */
-    public abstract void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result);
+    public /*abstract*/ void parseAssociatedNames(Job<?,?> job, Collection<GitHubRepositoryName> result) {
+        if (Util.isOverridden(GitHubRepositoryNameContributor.class, getClass(),
```

Comment:
I just followed the same pattern used by others, for example here: https://github.com/jenkinsci/rebuild-plugin/blob/e4fdcee240b609fe6bbbcd58071951d6b38c9453/src/main/java/com/sonyericsson/rebuild/RebuildValidator.java#L46